### ISSUE TITLE
Using required namespace directly as object in Cljs causes warning about unused require

### ISSUE THREAD

**Issue by Deraen** [borkdude/clj-kondo#446]
**version**

clj-kondo v2019.08.22-alpha-SNAPSHOT (2e452ff)

**platform**

Linux

**problem**

In ClojureScript, when namespace is required and bound to a name using `:as` the bound name can be used to directly access the object which namespace points to. This is often used with JS libraries (either with foreign-libs using`:global-exports`, or node packages with shadow-cljs or npm-deps) where the JS library exports a single JS object.

Currently Clj-kondo doesn't correctly detect uses of these objects and shows warning about unused require & info about unresolved object:

**repro**

```
clj-kondo --lint - <<EOF
(ns foo
  (:require [reagent.core :as r]
            ["apsl-react-native-button" :as NativeButton]))

(def button (r/adapt-react-class NativeButton))
EOF

```

**expected behavior**

Use of the require is correctly detected and no warning is shown.

**actual behavior**

```
<stdin>:3:14: warning: namespace apsl-react-native-button is required but never used
<stdin>:5:34: info: unresolved symbol NativeButton
linting took 9ms, errors: 0, warnings: 1
```

**Comment by borkdude** [comment]
Thanks! I'll close this after the next release, just as a reminder for myself to write a release note about it.

**Comment by borkdude** [comment]
What about:

```
cljs.user=> (require '[clojure.set :as s])
nil
cljs.user=> s
            ^
WARNING: Use of undeclared Var cljs.user/s at line 1
```

I had the behavior you implemented before at some point, but then users came with the message that when you did use s somewhere, it wasn't reported as unresolved anymore.
So is this behavior specific for shadow-cljs maybe?

**Comment by borkdude** [comment]
The relevant commit for this is: https://github.com/borkdude/clj-kondo/commit/18d97a6c3ca7d5d9eba351aa4291fc828bd2b215
and the issue was https://github.com/borkdude/clj-kondo/issues/374.

**Comment by borkdude** [comment]
As discussed on Slack: it would be nice to only have this behavior if the user requires the namespace using a string.
